lib/src/stat.dart

```diff
@@ -21,9 +21,11 @@ void mockGetModificationTime(MockTimeCallback callback) {
 }
 
 /// Gets the modification time for the file at [path].
-Future<DateTime> modificationTime(String path) async {
-  if (_mockTimeCallback != null) {
-    return _mockTimeCallback(path);
+/// Completes with `null` if the file does not exist.
+Future<DateTime?> modificationTime(String path) async {
+  var mockTimeCallback = _mockTimeCallback;
+  if (mockTimeCallback != null) {
+    return mockTimeCallback(path);
```

Comment:
[optional] I think this could also be `_mockTimeCallback?.call(path)`

Reply:
ACK, I think this will work.
I myself don't have conviction which pattern is better here - add a local variable, or force nullability.
Performance wise using a local variable might be a tiny bit faster, I think.